lib/katello/repository_types/README.md

```diff
@@ -28,24 +60,45 @@ Katello::RepositoryTypeManager.register(::Katello::Repository::YUM_TYPE) do
 ```
 This declares the the repository type, specifying a type identifier (e.g. 'yum')
 
-### Service class
+### Service class and friends
 
 ```ruby
-  service_class Katello::Pulp::Repository::Yum
+  pulp3_service_class Katello::Pulp3::Repository::Yum
+  pulp3_api_class Katello::Pulp3::Api::Yum
+  pulp3_plugin 'rpm'
+  ...
 ```
 
-Identifies the service class that is used to interface with pulp 2 and includes methods for creating and updating the type of repository
+Identifies the service class that is used to interface with Pulp and includes methods for creating and updating the type of repository. Also notes the location of the main API class and the name of the corresponding Pulp plugin.
+There are a number of other classes that need to be defined, so follow the example above to ensure that the new type has
+all the neccessary information.
  
  ### Content Type
  
 ```ruby
-  content_type Katello::Rpm, :priority => 1, :pulp2_service_class => ::Katello::Pulp::Rpm
+  default_managed_content_type Katello::Rpm::CONTENT_TYPE
+  content_type Katello::Rpm,
+    :priority => 1,
+    :pulp3_service_class => ::Katello::Pulp3::Rpm,
+    :primary_content => true,
+    :removable => true,
+    :uploadable => true
 ```
 
-Defines a content type to be used by the repository.  The first argument is the Active Record model class representing the content type.  This should extend the `Katello::Concerns::PulpDatabaseUnit` class.
+Firstly, default_managed_content_type sets the Rpm content type as the "default". The default managed type
+will be the type acted upon in places where it is not clear which content unit should be acted on. For example, during yum uploads,
+it will be assumed that the upload is of type Rpm unless otherwise specified.
```

Comment:
```suggestion
The `default_managed_content_type` method sets the Rpm content type as the default content type, which
the repository type will act upon when another content type is not explicitly declared. For example, a yum upload
will assume that the uploaded content is of Rpm type unless otherwise specified.
```

Trying to maintain the precision of what you have while improving the flow a bit, which is not easy. Feel free to disregard

Reply:
Yeah most of this is copy/pasted from the old readme that was accidentally deleted in an older commit.